### PR TITLE
Added documentation in `cirq_superstaq`

### PR DESCRIPTION
**Pull request by KabirDubey**
1. I put a docstring for the class initialization function for `CompilerOutput`. This may not be a standard, feel free to remove.

2. The exceptions in the counts function for `job.py` may be incorrect. Is the `RuntimeError` supposed to be a `SuperstaqUnsuccessfulJobException`?

3. The checks pass, but return:

```
************* Module cirq_superstaq.compiler_output
cirq-superstaq/cirq_superstaq/compiler_output.py:61:4: C0199: First line empty in method docstring (docstring-first-line-empty)
```

not sure if this is a problem (see 1).

### CODE REVIEW

**Review by dowusu-antwi — COMMENTED** [review]
Note: worked with Kabir offline to address commit issues (this branch was checked out from `update-basics` rather than `main`).

For posterity, fixing this can be done by `git checkout main -- <insert-file>` to reset the file (or `git rm <insert-file>` if the first command errors out).

**Comment by KabirDubey** [comment]
@PalashGoiporia, here are the files on the spreadsheet that this creates docstrings for:

./cirq-superstaq/cirq_superstaq: |
compiler_output.py | Kabir
job.py | Kabir
sampler.py | Kabir
serialization.py | Kabir
service.py | Kabir

These last three files on the spreadsheet required no docstrings.

File: cirq-superstaq/cirq_superstaq/compiler_output.py [hunk 1]
```diff
@@ -157,6 +171,8 @@ def read_json_aqt(  # pylint: disable=missing-param-doc
         json_dict: a JSON dictionary matching the format returned by /aqt_compile endpoint
         circuits_is_list: bool flag that controls whether the returned object has a .circuits
             attribute (if True) or a .circuit attribute (False)
+        num_eca_circuits: number of logically equivalent random circuits to generate for each
+            input circuit
```
Comment (dowusu-antwi) [inline cirq-superstaq/cirq_superstaq/compiler_output.py:175]:
```suggestion
        json_dict:  JSON dictionary matching the format returned by aqt_compile endpoint.
        circuits_is_list: Bool flag that controls whether the returned object has a .circuits
            attribute (if True) or a .circuit attribute (False).
        num_eca_circuits: Number of logically equivalent random circuits to generate for each
            input circuit.
```

File: cirq-superstaq/cirq_superstaq/compiler_output.py [hunk 2]
```diff
@@ -157,6 +171,8 @@ def read_json_aqt(  # pylint: disable=missing-param-doc
...
     Returns:
         a CompilerOutput object with the compiled circuit(s). If qtrl is available locally,
```
Comment (dowusu-antwi) [inline cirq-superstaq/cirq_superstaq/compiler_output.py:177]:
```suggestion
        A CompilerOutput object with the compiled circuit(s). If qtrl is available locally,
```

**Review by dowusu-antwi — COMMENTED** [review]
Nice -- just a few comments:
- descriptions [should be capitalized and end with a period](https://github.com/SupertechLabs/server-superstaq/blob/main/docs/source/developer_guide.rst#docstrings) (a relatively new standard for us)
- types are already documented in the argument list
- descriptions can be as long as we need, so I'd recommend not using abbreviations

File: cirq-superstaq/cirq_superstaq/compiler_output.py [hunk 3]
```diff
@@ -154,11 +167,13 @@ def read_json_aqt(  # pylint: disable=missing-param-doc
     """Reads out returned JSON from SuperstaQ API's AQT compilation endpoint.
 
     Args:
-        json_dict: a JSON dictionary matching the format returned by /aqt_compile endpoint
-        circuits_is_list: bool flag that controls whether the returned object has a .circuits
-            attribute (if True) or a .circuit attribute (False)
+        json_dict:  JSON dictionary matching the format returned by aqt_compile endpoint.
```
Comment (dowusu-antwi) [inline cirq-superstaq/cirq_superstaq/compiler_output.py:170]: nit: stray whitespace
```suggestion
        json_dict: JSON dictionary matching the format returned by aqt_compile endpoint.
```

File: cirq-superstaq/cirq_superstaq/compiler_output.py [hunk 4]
```diff
@@ -70,6 +71,18 @@ def __init__(
         jaqal_programs: Optional[Union[List[str], str]] = None,
         pulse_lists: Optional[Union[List[List[List[Any]]], List[List[List[List[Any]]]]]] = None,
     ) -> None:
+        """Initializes class attributes.
+
+        Args:
+            circuits: A list of at most 2 dimensions containing cirq circuit objects.
```
Comment (bharat-thotakura) [inline cirq-superstaq/cirq_superstaq/compiler_output.py:77]: minor nit:

```suggestion
            circuits: A list (of at most 2 dimensions) containing `cirq.Circuit` objects.
```

File: cirq-superstaq/cirq_superstaq/compiler_output.py [hunk 5]
```diff
@@ -70,6 +71,18 @@ def __init__(
...
+            final_logical_to_physicals: Post-compilation mapping of logical qubits to physical
+                qubits.
+            pulse_sequences: Qiskit pulse schedules for the compiled circuit(s).
+            seq: Qtrl pulse sequence, if qtrl is avaialable locally.
```
Comment (bharat-thotakura) [inline cirq-superstaq/cirq_superstaq/compiler_output.py:81]: typo 

```suggestion
            seq: Qtrl pulse sequence, if qtrl is available locally.
```

**Review by bharat-thotakura — APPROVED** [review]
LGTM

**Pull request merged**